src/model_repository_manager.cc

```diff
@@ -1803,27 +1818,57 @@ ModelRepositoryManager::Poll(
     // [DLIS-3487] Model doesn't need to be in repository if model files
     // are provided in flight.
     for (const auto& model : models) {
+      // Check model mapping first to see if matching model to load.
       bool exists = false;
-      for (const auto repository_path : repository_paths_) {
-        bool exists_in_this_repo = false;
-        const auto full_path = JoinPath({repository_path, model.first});
-        Status status = FileExists(full_path, &exists_in_this_repo);
-        if (!status.IsOk()) {
-          LOG_ERROR << "failed to poll model repository '" << repository_path
-                    << "' for model '" << model.first
-                    << "': " << status.Message();
+      auto model_it = model_mappings_.find(model.first);
+      if (model_it != model_mappings_.end()) {
+        auto res =
+            model_to_repository.emplace(model.first, model_it->second.first);
+        if (res.second) {
+          exists = true;
+        } else {
+          exists = false;
+          model_to_repository.erase(res.first);
+          LOG_ERROR << "failed to poll model '" << model.first
+                    << "': not unique across all model repositories";
           *all_models_polled = false;
-        } else if (exists_in_this_repo) {
-          auto res = model_to_repository.emplace(model.first, repository_path);
-          if (res.second) {
-            exists = true;
-          } else {
-            exists = false;
-            model_to_repository.erase(res.first);
-            LOG_ERROR << "failed to poll model '" << model.first
-                      << "': not unique across all model repositories";
+        }
```

Comment:
I don't think you need the duplication check here, since `model_mappings_` ensures the model name is mapped to one directory and we decided not to poll other repos if we have found a mapping, duplication can't happen.

However, you should have `FileExists()` check as below to make sure the mapping is valid, i.e. user may delete the model directory after registering the repo.

Reply:
Updated.